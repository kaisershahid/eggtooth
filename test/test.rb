require_relative '../lib/eggtooth.rb'

framework = Eggtooth::Framework.get_instance('test', {:root => '.', :log_std => true})
puts Eggtooth::PATH_INSTALL
puts "framework.id=#{framework.id}"

framework.startup

def crawl_resource(res)
	puts "-- crawling: #{res.path}"
	q = []
	res.children.each do |child|
		puts "   > #{child.name} : #{child.properties}"
		q << child.name if child.type != 'fs:file'
	end
	
	q.each do |cname|
		crawl_resource(res.child(cname))
	end
end

resman = framework.resource_manager
ext = resman.resolve('/ext')
dummy = Eggtooth::ResourceManager::NonExistingResource.new('/ext/dummy', nil, {'egg:type' => 'whatever', 'prop' => 'val'})
puts dummy.properties.inspect
puts dummy.properties.class
ext.editor.add(dummy)

pi_get = resman.path_info('/node.sel1.sel2.ext', 'GET')
pi_post = resman.path_info('/node.sel1.sel2.ext', 'POST')
puts "***" + pi_get.inspect
#puts "***" + pi_get.modify({:delete => ['selectors', 'resource']}).inspect
# actman = framework.service_manager.get_by_sid 'action.manager'
# action = actman.map(pi_get)
# action.exec(Eggtooth::Client::Request.new({}, pi_get, {}), nil)
# action.exec(Eggtooth::Client::Request.new({}, pi_post	, {}), nil)

# ext_sys = resman.resolve('/ext')
# puts ext_sys.class.to_s + " >> " + ext_sys.inspect
# crawl_resource(ext_sys)
# puts '***********'
# ext_sys = resman.resolve('sys')
# puts ext_sys.class.to_s + " >> " + ext_sys.inspect
# crawl_resource(ext_sys)

# puts '***********'
# ext_sys = resman.resolve('/merged/sys')
# puts ext_sys.class.to_s + " >> " + ext_sys.inspect
# crawl_resource(ext_sys)

# pathinfo = resman.path_info('/ext/sys.sel1.sel2.ext/suffix')
# puts pathinfo.inspect
# puts '**** action.default_rank check'
# puts Eggtooth::ActionManager::Action.default_rank(pathinfo, nil, ['sel1.sel2'], 'ext', nil)
# puts Eggtooth::ActionManager::Action.default_rank(pathinfo, nil, ['sel1'], 'ext', nil)
# puts Eggtooth::ActionManager::Action.default_rank(pathinfo, nil, ['sel1.sel2'], 'ext', '/suffix')
# puts Eggtooth::ActionManager::Action.default_rank(pathinfo, 'sys:folder', ['sel1.sel2'], 'ext', '/suffix')

#comp = Eggtooth::ActionManager::ScriptAction::EggshellCompiler.new(framework.service_manager, {:root => '${eggtooth.home}/var'})