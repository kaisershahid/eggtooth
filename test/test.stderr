I, [2017-01-11T11:58:33.802802 #86684]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:00:05.551191 #86756]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:20:35.479614 #87097]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:22:59.199952 #87181]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:24:23.946088 #87259]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:25:31.448960 #87330]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:25:33.285851 #87389]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:28:48.279661 #87486]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:29:08.709757 #87550]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:29:19.084347 #87611]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:29:40.971845 #87676]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:30:28.653302 #87743]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:32:17.803520 #87818]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:32:42.191193 #87883]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:36:07.139876 #87983]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:36:19.045327 #88045]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
I, [2017-01-11T12:36:46.124648 #88109]  INFO : << framework: test => /Users/kaisershahid/work/acmedinotech.com/git/ruby/eggtooth/test/. >>
